pkg/exec/exec.go

```diff
@@ -37,41 +38,63 @@ var (
 
 func RunCommandInOperatorPod(ctx *k8sutil.Context, cmd string, args []string, operatorNamespace, clusterNamespace string) string {
 
-	pod, err := k8sutil.WaitForOperatorPod(ctx, operatorNamespace)
+	pod, err := k8sutil.WaitForPodToRun(ctx, operatorNamespace, "app=rook-ceph-operator")
 	if err != nil {
 		os.Exit(1)
 	}
 
-	output := new(bytes.Buffer)
+	var stdout, stderr bytes.Buffer
 
-	ExecCmdInPod(ctx, cmd, pod.Name, "rook-ceph-operator", pod.Namespace, clusterNamespace, args, output)
-	return output.String()
+	err = execCmdInPod(ctx, cmd, pod.Name, "rook-ceph-operator", pod.Namespace, clusterNamespace, args, &stdout, &stderr)
+	if err != nil {
+		log.Error(err)
+	}
+	fmt.Println(stderr.String())
+	return stdout.String()
 }
 
-func RunShellCommandInOperatorPod(ctx *k8sutil.Context, arg []string, operatorNamespace, clusterNamespace string) string {
-	pod, err := k8sutil.WaitForOperatorPod(ctx, operatorNamespace)
+func RunCommandInToolboxPod(ctx *k8sutil.Context, cmd string, args []string, clusterNamespace string) string {
+	pod, err := k8sutil.WaitForPodToRun(ctx, clusterNamespace, "app=rook-ceph-tools")
 	if err != nil {
 		os.Exit(1)
 	}
 
-	cmd := "/bin/sh"
-	args := []string{"-c"}
-	args = append(args, arg...)
+	var stdout, stderr bytes.Buffer
 
-	output := new(bytes.Buffer)
-
-	ExecCmdInPod(ctx, cmd, pod.Name, "rook-ceph-operator", pod.Namespace, clusterNamespace, args, output)
-	return output.String()
+	err = execCmdInPod(ctx, cmd, pod.Name, "rook-ceph-tools", pod.Namespace, clusterNamespace, args, &stdout, &stderr)
+	if err != nil {
+		log.Error(err)
+	}
+	fmt.Println(stderr.String())
+	return stdout.String()
 }
 
-// ExecCmdInPod exec command on specific pod and wait the command's output.
-func ExecCmdInPod(ctx *k8sutil.Context, command, podName, containerName, podNamespace, clusterNamespace string, args []string, stdout io.Writer) {
+func RunCommandInLabeledPod(ctx *k8sutil.Context, label, container, cmd string, args []string, clusterNamespace string) string {
+	opts := metav1.ListOptions{LabelSelector: label}
+	list, err := ctx.Clientset.CoreV1().Pods(clusterNamespace).List(context.TODO(), opts)
+	if err != nil || len(list.Items) == 0 {
+		log.Error("failed to get rook mon pod where the command could be executed")
+		log.Fatal(err)
+	}
+	var stdout, stderr bytes.Buffer
+
+	err = execCmdInPod(ctx, cmd, list.Items[0].Name, container, list.Items[0].Namespace, clusterNamespace, args, &stdout, &stderr)
+	if err != nil {
+		log.Error(err)
```

Comment:
exit?

Reply:
actually, no because we want to retry running the command instead returning first time.